### DISCUSSION TITLE
Why do sdk.props and resulting .version file have different SDK versions?

### DISCUSSION THREAD

**Discussion by mateusrodrigues** [dotnet/source-build#3642] [Q&A] [answered]
The output of `dotnet --info` in a .NET 8 RC1 installation displays the following SDK version:
```
.NET SDK:
 Version:   8.0.100-rc.1.23455.1
 Commit:    e14caf947f

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  23.10
 OS Platform: Linux
 RID:         ubuntu.23.10-x64
 Base Path:   /usr/lib/dotnet/sdk/8.0.100-rc.1.23455.1/
...
```
By looking at the source, I see that this information comes from `<DotnetBasePath>/sdk/{version}/.version`.

However, the source-build source tree that generates this SDK has the following versioning information in `prereqs/git-info/sdk.props`:
```
<?xml version="1.0" encoding="utf-8"?>
<Project>
  <PropertyGroup>
    <GitCommitHash>113ba90fd02bb7241a8b4ef2a23d13cbd908cd19</GitCommitHash>
    <OfficialBuildId>20230905.21</OfficialBuildId>
    <OutputPackageVersion>8.0.100-rc.1.23455.21</OutputPackageVersion>
    <PreReleaseVersionLabel>rc.1</PreReleaseVersionLabel>
    <IsStable>false</IsStable>
  </PropertyGroup>
</Project>
```
Notice that the version from the props file ends in `.21`, while the former ends in `.1`. Why is that?

I also realized that the git commit hash in `dotnet --info`, under the SDK version, is the installer git hash, same as the one from `installer.props`. Is this by design?

**Comment by mthalman** [comment, marked as answer]
This is related to https://github.com/dotnet/source-build/issues/3521.

> I also realized that the git commit hash in dotnet --info, under the SDK version, is the installer git hash, same as the one from installer.props. Is this by design?

It would make more sense if this was the commit from the VMR. Would you like to open an issue for that?

**Reply by mateusrodrigues** [reply]
Sure can. And thanks for the versioning issue pointer!

**Reply by mateusrodrigues** [reply]
Opened issue #3643.
